Wikipedia/Code/Article as a Living Document/ArticleAsLivingDocReferenceCollectionViewCell.swift

```diff
@@ -0,0 +1,100 @@
+import UIKit
+
+class ArticleAsLivingDocReferenceCollectionViewCell: ArticleAsLivingDocHorizontallyScrollingCell {
+    
+    private let titleLabel = UILabel()
+    private var reference: ArticleAsLivingDocViewModel.Event.Large.Reference? = nil
+    private var iconTitleBadge: IconTitleBadge?
+    
+    override func sizeThatFits(_ size: CGSize, apply: Bool) -> CGSize {
+        
+        let adjustedMargins = ArticleAsLivingDocViewModel.Event.Large.padding
+        
+        var availableTitleWidth = size.width - adjustedMargins.left - adjustedMargins.right
+        let availableDescriptionWidth = availableTitleWidth
+        
+        if let iconTitleBadge = iconTitleBadge {
+            let maximumBadgeWidth = min(size.width - adjustedMargins.right - adjustedMargins.left, size.width / 3)
+            let iconBadgeOrigin = CGPoint(x: size.width - adjustedMargins.right, y: adjustedMargins.top)
+            let iconBadgeFrame = iconTitleBadge.wmf_preferredFrame(at: iconBadgeOrigin, maximumWidth: maximumBadgeWidth, alignedBy: .forceLeftToRight, apply: false)
+            if (apply) {
+                let iconTitleBadgeX = size.width - adjustedMargins.right - iconBadgeFrame.width
+                iconTitleBadge.frame = CGRect(x: iconTitleBadgeX, y: iconBadgeFrame.minY, width: iconBadgeFrame.width, height: iconBadgeFrame.height)
+            }
+            let titleBadgeSpacing = CGFloat(10)
+            availableTitleWidth -= iconBadgeFrame.width + titleBadgeSpacing
+        }
+        
+        let titleOrigin = CGPoint(x: adjustedMargins.left, y: adjustedMargins.top)
+        let titleFrame = titleLabel.wmf_preferredFrame(at: titleOrigin, maximumWidth: availableTitleWidth, alignedBy: .forceLeftToRight, apply: apply)
+        
+        let titleDescriptionSpacing = ArticleAsLivingDocViewModel.Event.Large.changeDetailReferenceTitleDescriptionSpacing
+        let descriptionOrigin = CGPoint(x: adjustedMargins.left, y: titleFrame.maxY + titleDescriptionSpacing)
+        
+        let _ = descriptionTextView.wmf_preferredFrame(at: descriptionOrigin, maximumWidth: availableDescriptionWidth, alignedBy: .forceLeftToRight, apply: apply)
+        
+        let finalSize = CGSize(width: size.width, height: size.height)
```

Comment:
Am I correct in understanding: `finalSize` is the `size` originally passed in? Could we just `return size`?

Reply:
Yeah, I think we can. Some of this messiness is holdover from back when I had each cell sizing to a different height, but at this point each side scrolling cell can return the original size passed in at the end. I'll do this when I do my sizing pass cleanup stuff from my other PR feedback.